plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt

```diff
@@ -57,15 +57,15 @@ private tailrec fun searchEditorConfigFiles(
     fileCollection: FileCollection
 ): FileCollection {
     val editorConfigFC = projectPath.resolve(EDITOR_CONFIG_FILE_NAME)
-    val outputCollection = if (editorConfigFC != null) {
+    val outputCollection = if (editorConfigFC.toFile().exists()) {
```

Comment:
Is this evaluated eagerly or at task execution time?

Reply:
I think eagerly on task configuration phase

Reply:
Is that a problem? Do we care if the file is added/modified by a previous task?

Reply:
I would say - we care, see, for example, #210 discussion.

What is the best way to do it? Current implementation: https://github.com/JLLeitschuh/ktlint-gradle/blob/0b019e0c0d99c2a34c2ff40fef61a1fefc21eebf/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt#L42
 
Use `by lazy { .. }` at this line or use `ObjectFactory.property { .. }`?

Reply:
Make it an input to the task that's a `FileProperty`? Honestly, I'd have to dig into where this code is actually called and see why you're adding it to this particular `FileCollection` to understand better what's going on.

Reply:
I think using `by lazy { ... }` will work here.

Reply:
Tested, works good - `getEditorconfigFiles()` is only evaluated on task execution when Gradle gets inputs for the task.